### PR TITLE
Support AVC3

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: lib/membrane_mp4/demuxer/isom.ex [hunk 1]
```diff
@@ -40,7 +40,7 @@ defmodule Membrane.MP4.Demuxer.ISOM do
       any_of(
         %Membrane.AAC{config: {:esds, _esds}},
         %Membrane.H264{
-          stream_structure: {:avc1, _dcr},
+          stream_structure: {_avc, _dcr},
```
Comment (varsill) [inline lib/membrane_mp4/demuxer/isom.ex:43]: We should assert on `_avc` being either `:avc1` or `:avc3`

File: lib/membrane_mp4/muxer/isom.ex [hunk 2]
```diff
@@ -96,25 +96,28 @@ defmodule Membrane.MP4.Muxer.ISOM do
...
-      # when stream format is duplicated - ignore
-      ctx.pads[pad].stream_format == stream_format ->
+      # Handle receiving a stream format of the same type
+      %^type{} ->
         state
 
       # otherwise we can assume that output will be corrupted
-      true ->
-        raise "ISOM Muxer doesn't support variable parameters"
+      previos_format ->
```
Comment (varsill) [inline lib/membrane_mp4/muxer/isom.ex:115]:
```suggestion
      previous_format ->
```

File: lib/membrane_mp4/muxer/isom.ex [hunk 3]
```diff
@@ -96,25 +96,28 @@ defmodule Membrane.MP4.Muxer.ISOM do
...
+      nil ->
         update_in(state, [:pad_to_track, pad_ref], fn track_id ->
           Track.new(track_id, stream_format, state.chunk_duration)
         end)
 
-      # Handle receiving all but the first stream format on the given pad,
-      # when stream format is duplicated - ignore
-      ctx.pads[pad].stream_format == stream_format ->
+      # Handle receiving a stream format of the same type
+      %^type{} ->
```
Comment (varsill) [inline lib/membrane_mp4/muxer/isom.ex:111]: Is it OK for us to accept change from `:avc3` to `:avc1`? Wouldn't it require appending something to the DCR?

File: test/membrane_mp4/muxer/isom/integration_test.exs [hunk 4]
```diff
@@ -210,58 +228,4 @@ defmodule Membrane.MP4.Muxer.ISOM.IntegrationTest do
...
-      assert_receive {:DOWN, ^monitor_ref, :process, ^pid,
-                      {:membrane_child_crash, :muxer,
-                       {%RuntimeError{
-                          message: "ISOM Muxer doesn't support variable parameters"
-                        }, _stacktrace}}},
-                     1_000
-    end
-  end
-
-  describe "ctts table" do
```
Comment (varsill) [inline test/membrane_mp4/muxer/isom/integration_test.exs:241]: Why have you removed this test?

**Review by varsill — CHANGES_REQUESTED** [review]

File: test/membrane_mp4/muxer/isom/integration_test.exs (hunk 4 above)
Comment (mat-hek) [reply test/membrane_mp4/muxer/isom/integration_test.exs:241]: ups :P

File: lib/membrane_mp4/muxer/isom.ex [hunk 5]
```diff
@@ -96,27 +96,36 @@ defmodule Membrane.MP4.Muxer.ISOM do
...
+      {stream_format, stream_format} ->
         state
 
-      # otherwise we can assume that output will be corrupted
-      previos_format ->
+      {%Membrane.H264{stream_structure: {:avc3, _dcr1}},
+       %Membrane.H264{stream_structure: {:avc3, _dcr2}}} ->
+        state
+
+      {%Membrane.H265{}, %Membrane.H265{}} ->
```
Comment (varsill) [inline lib/membrane_mp4/muxer/isom.ex:120]: I believe you should check for: 
```suggestion
      {%Membrane.H265{stream_structure: {:hev1, _dcr1}}, %Membrane.H265{stream_structure: {:hev1, _dcr2}}} ->
```

**Review by varsill — CHANGES_REQUESTED** [review]

**Review by varsill — APPROVED** [review]
🥇

**Pull request merged**